src/serializer/ResidualHeapSerializer.js

```diff
@@ -206,18 +205,11 @@ export class ResidualHeapSerializer {
   // TODO: revisit this and fix additional functions to be capable of delaying initializations
   additionalFunctionValueNestedFunctions: Set<FunctionValue>;
   currentAdditionalFunction: void | FunctionValue;
-  functionNames: Map<FunctionValue, string>;
-
-  _getFunctionName(f: FunctionValue): string {
-    let n = this.functionNames.get(f);
-    if (n === undefined) this.functionNames.set(f, (n = this.functionNameGenerator.generate(f.__originalName || "")));
```

Comment:
The old function names were unique, not any more. Is that okay?

Reply:
_getFunctionName() is only used for debugging purpose so it's should be fine. 
This diff is just the first refactoring to improve it not the final/perfect one. 99% of the case it will not have duplicate, and user can always modify the original source to avoid duplicate for important debugging scenario if needed. In future diff, we may want to add the function's serialized identifier name together with original name.